### DISCUSSION TITLE
Expose Climate Activity Status (COOLING / HEATING / ...)

### DISCUSSION THREAD

**Discussion by brave0d** [bimmerconnected/bimmer_connected#576] [Feature request] [answered]
Now we have nice switch to turn on/off climate in the car. The problem is that car randomly enables COOLING or HEATING. Probably it tries to be clever and predict what I really wanted, but it happened couple times to me that it start cooling even if the outside temperature was quite low.

If we can expose ClimateActivityStatus, then we can add a new sensor and build an automation on that to turn off climate and turn it back on (it usually changes the mode).

Unless there is another way to do it? But anyway, exposing climate activity status could be beneficial.

Thanks!

**Comment by wischnu** [comment]
Hello,
I support this feature request. Today I had the same issue, where the car was cooling, despite -3 degree outside. I later say that in the app it was visible that it starting cooling instead of heating (blue lines). With this status value we would be able to build a work around. 

Many thanks.

**Reply by Debug1Bug** [reply]
Thank you for a great integration. It seems to be a known problem that the car randomly chooses the wrong mode. And yes, the chosen status is visible in the app - so hopefully possible to access through the API. I would really like to be able to stop and restart climatization, if the car is in the wrong mode. It seems from experiences on the internet that the car always chooses the right mode on the second try.

**Comment by rikroe** [comment]
This is dependend on https://github.com/home-assistant/core/pull/110287 / https://github.com/home-assistant/core/pull/110294 with no ETA, but at least the PRs have been created ;)

**Reply by rikroe** [reply, marked as answer]
Implemented as https://github.com/home-assistant/core/pull/110287 got merged.
